Review changes

RalucaMihaela opened the pull request.

Commit:
Added Simple Test 1

Commit:
Added Simple Test 2

Commit:
Rename file

Commit:
Rename file

Commit:
Removing ignored files

Commit:
Added Simple Test 3

RalucaMihaela committed:
Added the [2,2,2] test

RalucaMihaela committed:
Changed the code so that the test [2,2,2] to pass 

I changed the logic of the problem, I used an array to count how many times each number is seen. I know that in the problem description it is saying that I can not use another structure, but I do not know how to do this in other way.

RalucaMihaela committed:
Pulled out the verification if the array has only 1 item

RalucaMihaela committed:
Removed the condition if the array is bigger than 0

alexandru-calinoiu commented:
There is no point opening a new pr you can simply reply to the existing  pr or comment.
(no need to change this one now as  I closed the previous pr, will continue or review here)

levelboy commented:
Why did you use two different naming conventions? (e.g. someFile.js and some-file.js)

levelboy commented on simpleTest1/appFirstPb.js:
```diff
@@ -0,0 +1,19 @@
+    let notNumber = 0;
+
+    for (var index = 0; index < array.length; index++) {
+        if (parseInt(array[index]) === array[index]) {
```
Is there the possibility for items in the array to not be numbers?

RalucaMihaela replied:
The problem said that I have to do the sum of even numbers. I know that even/odd only refers to integers, so that is why I verified If I had only integers. An array can have different types of items, of string type, boolean type etc. On best-case scenario I think I can assume I have only integers.

levelboy commented on simpleTest1/appFirstPb.js:
```diff
@@ -0,0 +1,19 @@
+            notNumber++;
+        }
+    }
+    if (notNumber === array.length) {
```
Is this part of the problem statement?

levelboy commented on simpleTest1/test/appFirstPbTest.js:
```diff
@@ -0,0 +1,52 @@
+        var array = [1.2, 2, 3];
+        assert.equal(app(array),2);
+      });
+      it('Even and odd elements should return sum of even elements', function(){
```
This description is not what the setup does. The setup adds only even numbers.

RalucaMihaela replied:
Yes I see. My mistake, I did not see that I did not change the test array.

levelboy commented on simpleTest2/appSecondPb.js:
```diff
@@ -0,0 +1,22 @@
+                        	secondMAX  = array[index];
+                  }
+            } else { 	 
+                  secondMAX = 0;
```
Why not set them back to `Number.NEGATIVE_INFINITY` ?

RalucaMihaela replied:
Yes, in this part best way is to set the variables with the values with which they were are initialized, but I didn't like to see -INFINITY, that is why I assigned those two variables with 0.

levelboy commented on simpleTest2/appSecondPb.js:
```diff
@@ -0,0 +1,22 @@
+    if(array.length === 0) return 0;
+
+    for(var index = 0; index < array.length; index++) {
+            if(parseInt(array[index]) === array[index]) {
```
As with the first problem, it would be safe to assume that all that you receive are integers

RalucaMihaela replied:
Ok, I understand

levelboy commented on duplicates-in-array.js:
```diff
@@ -0,0 +1,23 @@
+    var length = array.length;
+    var counter = zeroFilledArray(array.length);
+
+    if (array.length === 0) return false;
```
You already have a variable for `array.length`

levelboy commented on duplicates-in-array.js:
```diff
@@ -0,0 +1,23 @@
+module.exports = function (array, n) {
+
+    var length = array.length;
+    var counter = zeroFilledArray(array.length);
```
You already have a variable for `array.length`

levelboy commented on duplicates-in-array.js:
```diff
@@ -0,0 +1,23 @@
+
+    if (array.length === 0) return false;
+    if (n != length) return false;
+    for (var index = 0; index < array.length; index++) {
```
You already have a variable for array.length

levelboy commented on duplicates-in-array.js:
```diff
@@ -0,0 +1,23 @@
+module.exports = function (array, n) {
```
Does the problem statement mention that you receive two parameters for this function?

RalucaMihaela replied:
No it does not say. As I said to Costel on his review, I used n to verify if my given array has the number of items I want and to verify if each item is smaller than n. I could have used only the array's length to verify if each number is smaller than the length, because in the description does not say that I have to check if the array is in correct form.

levelboy commented on duplicates-in-array.js:
```diff
@@ -0,0 +1,23 @@
+      if (array[index] < 1 || array[index] > n) {
+          return false;
+      } else {
+    		if(counter[array[index]] === 1){
```
Can you think of a method to not use the bit array to verify for duplicates?

RalucaMihaela replied:
At first I did this problem by verifying if the sum of all numbers is n(n+1)/2.  If it wasn't equal then I had duplicates, but then I saw that this case didn't work If I did not have the array as a series from 1 to n.  I will try to do it in another way.

RalucaMihaela commented:
> Why did you use two different naming conventions? (e.g. someFile.js and some-file.js)

After I did the first two problems, Alex recommended me to use javascript naming convention so I tried to use it only on the third problem, because I added it the last one. I will change the naming convetion for the others too.

RalucaMihaela committed:
Assume that the array has only integer numbers

RalucaMihaela committed:
Corrected the test for even and odd elements

RalucaMihaela committed:
Assume that array has only integers

RalucaMihaela committed:
Remove the test to check if I have float numbers

RalucaMihaela committed:
A method that does not use another structure to find duplicates

I will loop through the array and if I will find an element positive I'll make it negative. After that if I will find an element that is negative, I will say the element at index "index" in the array is a duplicate.

RalucaMihaela committed:
Rename appFirstPb.js to first-problem.js

RalucaMihaela committed:
Rename appSecondPb.js to second-problem.js

RalucaMihaela committed:
Rename appSecondPbTest.js to second-problem-test.js

levelboy commented on duplicates-test.js:
```diff
@@ -0,0 +1,49 @@
+const assert = require('assert');
+const app = require('../duplicates-in-array');
+
+describe('App', function() {
```
Tests don't seem to reflect the last modifications to the function since https://github.com/Agilefreaks/Test/pull/26/files#diff-9b7a88c0978b5162b6f6b6e273897839R1 has one parameter but you still call it with two.

levelboy commented on duplicates-in-array.js:
```diff
@@ -0,0 +1,14 @@
+module.exports = function (array){
+  for(var index = 0; index < array.length; index++){
+       var item = array[Math.abs(array[index])];
```
`array[Math.abs(array[index])]` shows up in three places in this method

levelboy commented on duplicates-in-array.js:
```diff
@@ -0,0 +1,14 @@
+module.exports = function (array){
+  for(var index = 0; index < array.length; index++){
+       var item = array[Math.abs(array[index])];
+       if (array.length === 0) return false;
```
Why is this `if` in the for loop?

levelboy commented on duplicates-in-array.js:
```diff
@@ -0,0 +1,14 @@
+module.exports = function (array){
+  for(var index = 0; index < array.length; index++){
```
You can iterate over the array with `forEach()`

levelboy reviewed (commented):
Indentation is all messed:
There are:
- 8 spaces
- 4 spaces
- 2 spaces
- 1 tab

Choose one and stick to it

levelboy commented:
Closing this. Thanks!
